### DISCUSSION TITLE
How to extract text and tables at the same time

### DISCUSSION THREAD

**Discussion by jfuruness** [jsvine/pdfplumber#632] [Q&A] [answered]
With the pdfplumber library, you can extract the text of a PDF page, or you can extract the tables from a pdf page.

The issue is that I can't seem to find a way to extract text __and__ tables. Essentially, if the pdf is formatted in this way:

```
text1
tablename
___________
| Header 1 |
------------
| row 1    |
------------

text 2
```

I would like the output to be:
```
["text 1",
 "table name",
 [["header 1"], ["row 1"]],
 "text 2"]
```

If there is no way to separate the text by line, this output is also fine:
```
["text 1\ntable name\n", [["header 1"], ["row 1"]], "text 2\n"]
```

In this example you could run extract_text from pdfplumber:
```
with pdfplumber.open("example.pdf") as pdf:
    for page in pdf.pages:
        page.extract_text()
```

but that extracts text and tables as text. You could run extract_tables, but that only gives you the tables. I need a way to extract both text and tables at the same time.

Is this built into the library some way that I don't understand? If not, is this possible?

Thank you so much for any you help you can provide, this library is awesome!

**Comment by jfuruness** [comment]
I also do see this issue, which explains how to extract text __without__ tables:
https://github.com/jsvine/pdfplumber/issues/242

But I want something slightly different - I want text __and__ tables, one after the other

**Reply by jsvine** [reply]
Hi @jfuruness, and for your kind words about pdfplumber! Glad that you're finding it useful. You are correct: This functionality is not currently built into pdfplumber. It's theoretically possible, though, and I'll consider it for future versions of the project.

In the meantime, you may be able to cobble together a solution between `page.find_tables(...)` and `page.filter(...)` (filtering out objects within the `table.bbox` properties of each table).

**Comment by jfuruness** [comment]
@jsvine thank you so much for your response! Currently I'm extracting the tables, then extracting the text, and then I replace the text with tables with a special function. It's pretty hacky though and probably prone to breakage, I'd imagine there's a better way of doing it. Thank you for your consideration for a possible future version, best of luck with everything! You can close this ticket

**Comment by jfuruness** [comment, marked as answer]
For anyone that comes across this in the future, I asked a similar question on stack overflow, and it was answered! Note that this probably only works when there are no words to the left and right of the table itself (this was my case).

https://stackoverflow.com/q/71612119/8903959
